### PR TITLE
Clarify the requirement about newly created Span

### LINKED ISSUE #1508: Clarification needed for Span Creation section of Tracer API
The [Span Creation](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.0.1/specification/trace/api.md#span-creation) says:

> Span creation MUST NOT set the newly created Span as the currently active Span by default, but this functionality MAY be offered additionally as a separate operation.

This requirement is more applicable to languages with implicit contexts but not suitable for languages with explicit contexts.

For some languages, like Go, we use explicit and immutable `Context` so the SDK returns a new `Context` with the new `Span` set as active. Users can choose which of the `Context` as the parent `Span` while using such `Context` instances to not cause an arbitrary `Span` becoming a parent.

Also, explicit set the currently active `Span` to `Context` seems wordy while the `Context` instance is explicit.
For instance:
```go
newCtx, span = CreateSpan(ctx, "span-name")
newCtx = SetActiveContext(newCtx, span)

newCtx2, span = CreateSpan(newCtx, "span-name")
newCtx2 = SetActiveContext(newCtx2, span)
```
Rather than:
```go
newCtx, span = CreateSpan(ctx, "span-name")

newCtx2, span = CreateSpan(newCtx, "span-name")
```

Prior conversations in [Slack](https://cloud-native.slack.com/archives/C01N7PP1THC/p1614737963034700)

### PR DESCRIPTION
**Pull request by XSAM**
The requirement that `Span` creation can't set the newly created `Span` in `Context` is not suitable for languages that use explicit `Context`, such as Go.

This corrects that ambiguity.

Resolves #1508

### CODE REVIEW

File: specification/trace/api.md [hunk 1]
```diff
@@ -328,9 +328,9 @@ directly. All `Span`s MUST be created via a `Tracer`.
 
 There MUST NOT be any API for creating a `Span` other than with a [`Tracer`](#tracer).
 
-`Span` creation MUST NOT set the newly created `Span` as the currently
-active `Span` by default, but this functionality MAY be offered additionally
-as a separate operation.
+`Span` creation MUST NOT set the newly created `Span` as the
+active `Span` in the implicit `Context` by default, but this functionality MAY be offered additionally
```
Comment (arminru) [inline specification/trace/api.md:332]:
```suggestion
active `Span` in the [implicit `Context`](#context-interaction) by default, but this functionality MAY be offered additionally
```

File: specification/trace/api.md [hunk 2]
```diff
@@ -328,9 +328,9 @@ directly. All `Span`s MUST be created via a `Tracer`.
...
+as a separate operation. Languages that use explicit `Context` are not suitable for this requirement.
```
Comment (arminru) [inline specification/trace/api.md:333]:
```suggestion
as a separate operation in languages with implicit context propagation.
```

**Review by arminru — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

File: specification/trace/api.md [hunk 3]
```diff
@@ -328,9 +328,9 @@ directly. All `Span`s MUST be created via a `Tracer`.
 
 There MUST NOT be any API for creating a `Span` other than with a [`Tracer`](#tracer).
 
-`Span` creation MUST NOT set the newly created `Span` as the currently
-active `Span` by default, but this functionality MAY be offered additionally
-as a separate operation.
+`Span` creation MUST NOT set the newly created `Span` as the
+active `Span` in the [implicit `Context`](#context-interaction) by default, but this functionality MAY be offered additionally
+as a separate operation in languages with implicit context propagation.
```
Comment (carlosalberto) [inline specification/trace/api.md:333]:
```suggestion
In languages with implicit `Context` propagation, `Span` creation MUST NOT
set the newly created `Span` as the active `Span` in the
[current `Context`](#context-interaction) by default, but this functionality
MAY be offered additionally as a separate operation.
```

Comment (carlosalberto) [reply specification/trace/api.md:333]: I think that this reads better, by **starting** the paragraph mentioning this case applies only for languages with implicit context handling.

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**